lib/internal/bootstrap_node.js

```diff
@@ -370,17 +369,17 @@
     const { kAfter, kExecutionAsyncId,
             kInitTriggerAsyncId } = async_wrap.constants;
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, shouldCatch = true) {
       var caught;
 
       // It's possible that kInitTriggerAsyncId was set for a constructor call
       // that threw and was never cleared. So clear it now.
       async_id_fields[kInitTriggerAsyncId] = 0;
 
-      if (process.domain && process.domain._errorHandler)
+      if (shouldCatch && process.domain && process.domain._errorHandler)
         caught = process.domain._errorHandler(er);
 
-      if (!caught)
+      if (shouldCatch && !caught)
```

Comment:
I'm still not sure what exactly `shouldCatch` entails. I'd like to get https://github.com/nodejs/node/pull/17159 done at some point... maybe we can try to do that before this lands?

Reply:
This part is slightly different than the work within that PR. It's so we don't need `ClearFatalExceptionHandlers` which deep reaches into a bunch of JS code (including `_events`).

The part of this PR that would be made obsolete by #17159 is within lib/domain.js (& a bit within `node.cc`). I'm guessing that could land and then we can refactor later? Not sure where that work is at though so I'm happy to wait if it's close.